packages/core/strapi/src/services/server/middleware.ts

```diff
@@ -36,15 +36,7 @@ const dummyMiddleware: Common.MiddlewareHandler = (_, next) => next();
  * Initialize every configured middlewares
  */
 const resolveMiddlewares = (
-  config: Array<
-    | string
-    | {
-        name?: string;
-        resolve?: string;
-        config?: unknown;
-      }
-    | Common.MiddlewareHandler
-  >,
+  config: Array<Common.MiddlewareName | Common.MiddlewareConfig | Common.MiddlewareHandler>,
```

Comment:
`Common.MiddlewareName`

Shouldn't this be a middleware unique ID instead?
They are not auto generated yet but we defined them in the shared registries & the UID namespace

Reply:
I don't want to break it since the docs say we allow any string so until we're sure we need to keep `string` like we used originally here